lms/djangoapps/ccx/tests/test_field_override_performance.py

```diff
@@ -113,17 +117,42 @@ def setup_course(self, size, enable_ccx):
             self.course.id
         )
 
-    def grade_course(self, course):
+        if enable_ccx:
+            self.ccx = CcxFactory.create()
+            CcxMembershipFactory.create(
+                student=self.student,
+                ccx=self.ccx
+            )
+
+    def grade_course(self, course, view_as_ccx):
         """
         Renders the progress page for the given course.
         """
+        course_key = course.id
+        if view_as_ccx:
+            course_key = CCXLocator.from_course_locator(course_key, self.ccx.id)
+
         return progress(
             self.request,
-            course_id=course.id.to_deprecated_string(),
+            course_id=unicode(course_key),
             student_id=self.student.id
         )
 
-    def instrument_course_progress_render(self, course_width, enable_ccx, queries, reads, xblocks):
+    def assertMongoCallCount(self, calls):
+        """
+        Assert that mongodb is queried ``calls`` times in the surrounded
+        context.
+        """
+        return check_mongo_calls(calls)
+
+    def assertXBlockInstantiations(self, instantiations):
+        """
+        Assert that exactly ``instantiations`` XBlocks are instantiated in
+        the surrounded context.
+        """
+        return check_sum_of_calls(XBlock, ['__init__'], instantiations, instantiations, include_arguments=False)
```

Comment:
It'd be good to take this method down into factories, named closed to this name. No?

Also, either way, if this technique could be added to this page:
https://openedx.atlassian.net/wiki/display/PLAT/How+to+Test+MongoDB+and+MySQL+Call+Counts
or one in the same location, that'd be great.